### PR TITLE
Stop parsing METADATA section for IdMap

### LINKED ISSUE #8500: Compiler Internal Error: Could not deserialize idmap.
### Discord username

farmaazon

### What type of issue is this?

Intermittent – Occurring irregularly

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

Sometimes when I open project, I get no updates from the engine (the nodes are gray and visualizations are stuck in "loading" state). In the PM log, there is an error `Compiler Internal Error: Could not deserialize idmap.`

Strangely, it seems to not depend on the file I use, because opening same project second time works just as fine.

### Expected behaviour

IDE opens and nodes are evaluated. No Compiler Internal Errors.

### How we can reproduce it?

I see it rarely, but it happens more than once. 

I saw it for this project once, for example: 
[New_Project_33.zip](https://github.com/enso-org/enso/files/13613365/New_Project_33.zip)

### Screenshots or screencasts

_No response_

### Logs

PM logs after I opened project once and it failed as described. Then I opened the same project second time and it worked.

[enso-project-manager-2023-12-08.0.log](https://github.com/enso-org/enso/files/13613357/enso-project-manager-2023-12-08.0.log)

### Enso Version

IDE built on commit e4b2b56a405ef5036a9351319112c53ade3aed47 with nightly PM (2023-12-08

### Browser or standalone distribution

Chrome

### Browser Version or standalone distribution

 120.0.6099.71

### Operating System

Linux

### Operating System Version

_No response_

### Hardware you are using

_No response_

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

We infer old IDs from module's IRs location, instead. This workarounds ill-constructed IdMaps at the start of the project.

If IDE or backend ends up with invalid ID map, I suppose application of edits might still fail.

### Important Notes

Closes #8500 by not doing the parsing at all. Since I couldn't figure why we had invalid metadata in the first place, this might still bite us at some point.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.

### CODE REVIEW

**Review by 4e6 — APPROVED** [review]

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala [hunk 1]
```diff
@@ -112,6 +112,8 @@ final class EnsureCompiledJob(
         applyEdits(new File(module.getPath)).map { changeset =>
           compile(module)
             .map { _ =>
+              // Side-effect: ensures that module's source is correctly initialized.
```
Comment (JaroslavTulach) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala:115]: What happens when this side-effect isn't performed?

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala [hunk 2]
```diff
@@ -347,37 +349,6 @@ final class EnsureCompiledJob(
     )
   }
 
-  type IDMap = Seq[(org.enso.data.Span, java.util.UUID)]
-  private def splitMeta(code: String): (String, IDMap, io.circe.Json) = {
-    import io.circe.Json
```
Comment (JaroslavTulach) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala:352]: Wouldn't this, by a chance, allow us to remove circe from some of our project dependencies?

**Review by JaroslavTulach — APPROVED** [review]
The "side-effect" trigger is suspicious. Without it, I'd say this would be a great PR!

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala (hunk 1 above)
Comment (hubertp) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala:115]: We get a failed execution `The Main.main source not found.`

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala (hunk 2 above)
Comment (hubertp) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala:352]: Nope. We use circe for parsing configuration files as well as parsing requests.

**Comment by hubertp** [comment]
> The "side-effect" trigger is suspicious. Without it, I'd say this would be a great PR!

Previously we were doing `module.getSource().getCharacters()` so the side-effect was implicit. I found it the hard way.

**Pull request merged**